src/macaron/slsa_analyzer/checks/build_as_code_check.py

```diff
@@ -280,24 +283,59 @@ def run_check(self, ctx: AnalyzeContext, check_result: CheckResult) -> CheckResu
                                 ] = f"{ctx.remote_path}@refs/heads/{ctx.branch_name}"
                                 predicate["invocation"]["configSource"]["digest"]["sha1"] = ctx.commit_sha
                                 predicate["invocation"]["configSource"]["entryPoint"] = config_name
-                            check_result["result_tables"] = [
+                            check_result["result_tables"].append(
                                 BuildAsCodeTable(
                                     build_tool_name=build_tool.name,
                                     ci_service_name=ci_service.name,
                                     deploy_command=deploy_kw,
                                 )
-                            ]
+                            )
                             return CheckResultType.PASSED
 
-            pass_msg = f"The target repository does not use {build_tool.name} to deploy."
-            check_result["justification"].append(pass_msg)
-            check_result["result_tables"] = [BuildAsCodeTable(build_tool_name=build_tool.name)]
+        pass_msg = f"The target repository does not use {build_tool.name} to deploy."
+        check_result["justification"].append(pass_msg)
+        check_result["result_tables"].append(BuildAsCodeTable(build_tool_name=build_tool.name))
+        return CheckResultType.FAILED
+
+    def run_check(self, ctx: AnalyzeContext, check_result: CheckResult) -> CheckResultType:
+        """Implement the check in this method.
+
+        Parameters
+        ----------
+        ctx : AnalyzeContext
+            The object containing processed data for the target repo.
+        check_result : CheckResult
+            The object containing result data of a check.
+
+        Returns
+        -------
+        CheckResultType
+            The result type of the check (e.g. PASSED).
+        """
+        # Get the build tool identified by the mcn_version_control_system_1, which we depend on.
+        build_tools = ctx.dynamic_data["build_spec"]["tools"]
+
+        if not build_tools:
+            check_result["result_tables"] = [BuildAsCodeTable()]
+            failed_msg = "The target repository does not have any build tools."
+            check_result["justification"].append(failed_msg)
             return CheckResultType.FAILED
 
-        check_result["result_tables"] = [BuildAsCodeTable()]
-        failed_msg = "The target repository does not have a build tool."
-        check_result["justification"].append(failed_msg)
-        return CheckResultType.FAILED
+        ci_services = ctx.dynamic_data["ci_services"]
+
+        # Check if "build as code" holds for each build tool.
+        for tool in build_tools:
+            res = self._check_build_tool(tool, ctx, ci_services, check_result)
+
+            # Divide up build tool outputs
+            check_result["justification"].append("\n")
```

Comment:
I don't think we need to add a  "\n" character to divide up the outputs. The content of each element of `check_result["justification"]` is not used for rendering the reports. So if you add "\n" here, the string `"\n"` will appear instead of a new line.
Furthermore, in the HTML reports each element in `check_result["justification"]` is rendered as a separated bullet point in the justification column.